libraries/AP_BattMonitor/AP_BattMonitor_ADS1115.cpp

```diff
@@ -0,0 +1,57 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor.h"
+#include <AP_ADC/AP_ADC.h>
+#include "AP_BattMonitor_ADS1115.h"
+
+#define BATTMON_ADS1115_DEBUG 0
+
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_BattMonitor_ADS1115::AP_BattMonitor_ADS1115(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state): AP_BattMonitor_Backend(mon, mon_state)
+{    
+    _state.healthy = true;
+    _adc.init(ADS1115_I2C_BUS);
+}
+
+// read - read the voltage and current
+void
+AP_BattMonitor_ADS1115::read()
+{
+	// Just read through A0 and A1, and stop
+	// Voltage goes on A0 - 2
+	// Current goes on A1 - 3
+	adc_report_s report[6];
+	_adc.read(report, 6);
+
+    // get voltage and converts from ADC output to volts, adds offset to correct for actual voltage
+    _state.voltage = report[3].data / 100.0f + 0.3f;
+
+	// calculate time since last current read
+	uint32_t tnow = AP_HAL::micros();
+	float dt = tnow - _state.last_time_micros;
+    // get amperage - converts from weird 0.1 to correct ~0.004
+    _state.current_amps = -(report[2].data/200.0f);
+
+    // update total current drawn since startup
+    if (_state.last_time_micros != 0 && dt < 2000000.0f) {
+        // .0002778 is 1/3600 (conversion to hours)
+        // Change to division since compiled out and easier to read.
+        _state.current_total_mah += _state.current_amps * dt * (1.0/3600.0);
```

Comment:
Original multiplier was 0.0000002778f - see [HERE](https://github.com/Aerotenna/APM_OcPoC_Zynq/blob/mauch_battmon/libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp#L45)

I'm fairly certain the difference would account for the conversions from microseconds to seconds (*0.000001) and from amps to milliamps (*1000), so I think your total mAh calculation here is off by 1000. Check this again and see if it needs an edit.